src/validators/typed_dict.rs

```diff
@@ -198,9 +205,10 @@ impl Validator for TypedDictValidator {
                             Err(err) => return Err(err),
                         }
                     } else if let Some(ref default) = field.default {
-                        // TODO default needs to be copied here
+                        output_dict.set_item(&field.name_pystring, default)?;
+                    } else if let Some(ref default_factory) = field.default_factory {
                         output_dict
-                            .set_item(&field.name_pystring, default.as_ref(py))
+                            .set_item(&field.name_pystring, default_factory.call0(py)?)
```

Comment:
This is weird, in theory `default_factory.call0(py)?` would need a `.map_err(Into::<ValError>::into)?` too.

But it seems like because `Into` is implemented, we can probably remove all the `.map_err(Into::<ValError>::into)?` everywhere, I'll do it in a separate PR.

Reply:
Yes I think we can probably replace all the `.map_err(Into::<ValError>::into)?` by just `?`

Reply:
#170